core/src/main/java/net/opentsdb/query/plan/DefaultQueryPlanner.java

```diff
@@ -971,7 +971,8 @@ protected void setupPushDowns() {
       }
       
       final List<QueryNodeConfig> push_downs = Lists.newArrayList();
-      for (final QueryNodeConfig n : config_graph.predecessors(node)) {
+      final Set<QueryNodeConfig> nodes = Sets.newHashSet(config_graph.predecessors(node));
```

Comment:
why not iterate on the `config_graph.predecessors(node)` itself

Reply:
yeah, explicit dedupe to avoid a concurrent modification exception as the call below can mutate the graph.